### PR TITLE
Turns stamina regeneration modifier into a mob property

### PR DESCRIPTION
**Pull request by pali6**
## About the PR 
This PR turns stamina regeneration modifier into a mob property. This makes getting the current value of the modifier faster (it has been appearing in the profiler a bit) and also makes the code arguable clearer. Some minor mechanical differences might appear, who knows.

## Why's this needed? 
Performance etc.

### CODE REVIEW

**Comment by pali6** [comment]
Ah crap, secrets.

**Comment by pali6** [comment]
I made a secret submodule branch which will likely need to be merged manually and it'll be a bit of a mess. (Basically ping me if you are merging this and I'll fix it.)

File: code/WorkInProgress/ObjectProperties.dm [hunk 1]
```diff
@@ -537,4 +528,14 @@ to say if there's demand for that.
...
 
+/datum/objectProperty/equipment/stamregen
+	name = "Stamina regen."
+	id = "stamregen"
+	desc = "Affects stamina regenration." //Value is flat effective change to stamina regeneration.
+	tooltipImg = "stamregen.png"
+	defaultValue = 1
+	getTooltipDesc(var/obj/propOwner, var/propVal)
+		return "[propVal] stamina regen."
+	ASSOCIATE_MOB_PROPERTY(PROP_STAMINA_REGEN_BONUS)
```
Comment (ZeWaka) [inline code/WorkInProgress/ObjectProperties.dm:539]:
```suggestion
	defaultValue = 1
	
	getTooltipDesc(var/obj/propOwner, var/propVal)
		return "[propVal] stamina regen."

	ASSOCIATE_MOB_PROPERTY(PROP_STAMINA_REGEN_BONUS)
```

hiss

**Review by ZeWaka — APPROVED** [review]

**Comment by Tarmunora** [comment]
Any particular reason to to do stamina max at the same time?

**Comment by pali6** [comment]
> Any particular reason not to do stamina max at the same time?
This was surprisingly non-trivial (read not-directly-regexable) to do and only the stamina regen modifier popped up on the profiler.

**Pull request merged**